internal/lexer/token.go

```diff
@@ -0,0 +1,51 @@
+package lexer
+
+type TokenType int
+
+const (
+	EOF TokenType = iota
+	IDENTITY
+	NUMBER
+	STRING
+
+	PATH     // .
+	PIPE     // |
+	LBRACKET // [
+	RBRACKET // ]
+
+	beginOperatorSection
+	ASSIGN   // =
+	EQUALITY // ==
+	MERGE    // +
+	endOperatorSection
+)
+
+var tokens = []string{
+	EOF:      "EOF",
+	IDENTITY: "IDENTITY",
+	NUMBER:   "NUMBER",
+	STRING:   "STRING",
+
+	PATH:     "PATH",
+	PIPE:     "PIPE",
+	LBRACKET: "LBRACKET",
+	RBRACKET: "RBRACKET",
+
+	ASSIGN:   "ASSIGN",
+	EQUALITY: "EQUALITY",
+	MERGE:    "MERGE",
+}
+
+func (t TokenType) String() string {
+	return tokens[t]
+}
+
+type Token struct {
+	Position int
```

Comment:
Why do we need the position?

Reply:
It's needed for this https://github.com/nmstate/nmpolicy/pull/1/files#diff-3542290a91a71512da208c756566348f5684d9a86c634ca6102fa53afdb972ceR215-R217

Reply:
Please give me a high level (in your words) explanation for why it is needed.

Reply:
If a user create a NMPolicy with an invalid capture, we can just return an error saying "the capture with expression ... is invalid" or we can do better and say "the capture with expression ... is wrong at pos=3" or do even better since we have the position:
```
the capture with expression ... is wrong at pos=4
|foo.bar.dar
|....^
```